test/Digipolis.Auth.UnitTests/PDP/PolicyDecisionProviderTests.cs

```diff
@@ -106,26 +103,25 @@ public async Task ApiKeyIsSetInHeader()
             };
 
             var mockHandler = new MockMessageHandler<PdpResponse>(HttpStatusCode.OK, pdpResponse);
-            var provider = new PolicyDecisionProvider(mockedCache.Object, Options.Create(_options), mockHandler, _logger);
+            var httpClient = new HttpClient(mockHandler);
+            httpClient.DefaultRequestHeaders.TryAddWithoutValidation(HeaderKeys.Apikey, _options.PdpApiKey);
+            var uri = _options.PdpUrl.EndsWith("/") ? _options.PdpUrl : $"{_options.PdpUrl}/";
+            httpClient.BaseAddress = new Uri(uri);
+            var provider = new PolicyDecisionProvider(httpClient, mockedCache.Object, Options.Create(_options), _logger);
             var result = await provider.GetPermissionsAsync(_userId, _application);
 
-            Assert.Equal(_apiKey, mockHandler.Headers.GetValues(HeaderKeys.Apikey).FirstOrDefault());
+            Assert.Equal(_apiKey, httpClient.DefaultRequestHeaders.GetValues(HeaderKeys.Apikey).FirstOrDefault());
         }
 
         [Fact]
         public async Task ReturnsNullIfUserUnknown()
         {
             var mockedCache = CreateEmptyMockedCache();
-
-            var pdpResponse = new PdpResponse
-            {
-                applicationId = _application,
-                userId = _userId,
-                permissions = new List<String>(new string[] { requestedresource })
-            };
-
             var mockHandler = new MockMessageHandler<PdpResponse>(HttpStatusCode.NotFound, null);
-            var provider = new PolicyDecisionProvider(mockedCache.Object, Options.Create(_options), mockHandler, _logger);
+            var client = new HttpClient(mockHandler);
+            var uri = _options.PdpUrl.EndsWith("/") ? _options.PdpUrl : $"{_options.PdpUrl}/";
+            client.BaseAddress = new Uri(uri);
```

Comment:
Gelijkaardige opmerking: moet dat BaseAddress niet in de PolicyDecisionProvider gezet worden?

Reply:
Steekt ook in ServiceCollectionExtensions. HttpClients worden nu geinjecteerd, en niet meer geset in constructors. Ik kon vanuit de tests echter niet aan de servicecollectionextensions, dus heb headers & base address op deze manier moeten meegeven

Reply:
Steekt ook in ServiceCollectionExtensions. HttpClients worden nu geinjecteerd, en niet meer geset in constructors. Ik kon vanuit de tests echter niet aan de servicecollectionextensions, dus heb headers & base address op deze manier moeten meegeven